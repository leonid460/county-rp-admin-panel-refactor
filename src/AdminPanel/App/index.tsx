import React, {useEffect} from 'react';
import {
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import {useHistory} from 'react-router-dom';
import {observer} from 'mobx-react';
import GlobalStyle from './GlobalStyle';
import {StoreProvider} from 'AdminPanel/stores';
import {useStore} from 'AdminPanel/stores';
import Home from 'AdminPanel/components/pages/Home';
import AuthForm from 'AdminPanel/components/pages/Auth';
import Profile from 'AdminPanel/components/pages/Profile';
import Forum from 'AdminPanel/components/pages/Forum';
import Group from 'AdminPanel/components/pages/Group';
import CreateGroup from 'AdminPanel/components/pages/CreateGroup';
import EditGroup from 'AdminPanel/components/pages/EditGroup';
import Players from 'AdminPanel/components/pages/Players';
import EditPlayer from 'AdminPanel/components/pages/EditPlayer';
import CreatePlayer from 'AdminPanel/components/pages/CreatePlayer';
import Person from 'AdminPanel/components/pages/Person';
import EditPerson from 'AdminPanel/components/pages/EditPerson';
import Faction from 'AdminPanel/components/pages/Faction';
import CreateFaction from 'AdminPanel/components/pages/CreateFaction';
import EditFaction from 'AdminPanel/components/pages/EditFaction';
import AdminLevel from 'AdminPanel/components/pages/AdminLevel';
import EditAdminLevel from 'AdminPanel/components/pages/EditAdminLevel';
import CreateAdminLevel from 'AdminPanel/components/pages/EditAdminLevel';
import TestRoom from 'AdminPanel/components/pages/TestRoom';
import NotificationTray from 'AdminPanel/components/molecules/Notification';
import LogPage from 'AdminPanel/components/pages/Log';
import {routes} from 'AdminPanel/routes';

const App = observer(() => {
  useAuthCheck();

  return <>
    <GlobalStyle />
    <Switch>
      <Route exact path={routes.root}><Home /></Route>
      <Route path={routes.auth} component={AuthForm} />
      <Route path={routes.profile} component={Profile} />
      <Route path={routes.forum} component={Forum} />

      <Route exact path={routes.group} component={Group} />
      <Route path={routes.createGroup}><CreateGroup /></Route>
      <Route path={routes.editGroup + '/:id'}><EditGroup /></Route>

      <Route exact path={routes.players} component={Players} />
      <Route path={routes.editPlayer + '/:id'}><EditPlayer /></Route>
      <Route path={routes.createPlayer}><CreatePlayer /></Route>

      <Route exact path={routes.person} component={Person} />
      <Route path={routes.editPerson + '/:id'}><EditPerson /></Route>

      <Route exact path={routes.faction} ><Faction /></Route>
      <Route path={routes.createFaction}><CreateFaction /></Route>
      <Route path={routes.editFaction + '/:id'}><EditFaction /></Route>

      <Route exact path={routes.adminLevel} ><AdminLevel /></Route>
      <Route path={routes.createAdminLevel}><CreateAdminLevel /></Route>
      <Route path={routes.editAdminLevel + '/:id'}><EditAdminLevel /></Route>

      <Route exact path={routes.log}><LogPage /></Route>

      <Route path={routes.testRoom}><TestRoom/></Route>

      <Redirect to={routes.root} />
    </Switch>
    <NotificationTray />
  </>;
});

function useAuthCheck() {
  const {playerInfoStore} = useStore();
  const history = useHistory();
  const isAuthorized = playerInfoStore.isAuthorized;

  useEffect(() => {
    if (!isAuthorized && process.env.REACT_APP_DEV_MODE === 'false') {
      history.push('/admin/Auth');
    }
  }, [history, isAuthorized]);
}

export default () => <StoreProvider><App /></StoreProvider>;
