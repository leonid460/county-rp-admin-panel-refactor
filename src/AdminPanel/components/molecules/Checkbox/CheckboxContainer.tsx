import styled from 'styled-components';


const CheckboxContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  & input[type='checkbox'] {
    display: none;
    outline: none;
  }
`;


export default CheckboxContainer;
