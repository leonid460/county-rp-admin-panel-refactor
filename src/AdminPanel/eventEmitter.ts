type EventsStorage = {
  [key: string]: Function[];
}


class EventEmitter {
  events: EventsStorage;

  constructor() {
    this.events = {} as EventsStorage;
  }

  subscribe<T>(eventName: string, fn: (data: T) => void) {
    if (!this.events[eventName] ) {
      this.events[eventName] = [];
    }

    this.events[eventName].push(fn);

    return () => {
      this.events[eventName] = this.events[eventName].filter(eventFn => fn !== eventFn);
    };
  }

  emit<T>(eventName: string, data: T) {
    const event = this.events[eventName];

    if (event) {
      event.forEach(fn => {
        fn.call(null, data);
      });
    }
  }
}


export const eventEmitter = new EventEmitter();
