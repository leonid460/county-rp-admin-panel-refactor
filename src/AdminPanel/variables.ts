export const colors = {
  gray: '#8a8a8a',
  grayGradient: 'linear-gradient(180deg, #f3f3f3 0%, white 100%)',
  blueGradient: 'linear-gradient(180deg, #02149c 0%,#298ACF 100%)',
  redGradient: 'linear-gradient(180deg, #d10000 0%, #d27e7e 100%);',
  blue: '#0653c0',
};

export const zIndex = {

};

export const pagePadding = '50px';
