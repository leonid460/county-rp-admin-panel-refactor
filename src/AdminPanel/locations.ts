import {routes} from 'AdminPanel/routes';

type Location = {
  description: string;
  name: string;
  route: string;
};

export const locations: Location[] = [
  {
    name: 'Игроки',
    description: 'Описание',
    route: routes.players,
  },
  {
    name: 'Группы',
    description: 'Описание',
    route: routes.group,
  },
  {
    name: 'Персонажи',
    description: 'Описание',
    route: routes.person,
  },
  {
    name: 'Фракции',
    description: 'Описание',
    route: routes.faction,
  },
  {
    name: 'Админ уровни',
    description: 'Описание',
    route: routes.adminLevel,
  },
  {
    name: 'Логи',
    description: 'Тут лежат логи',
    route: routes.log,
  },
  {
    name: 'Тестовая комната',
    description: 'Ничего странного здесь не происходит',
    route: routes.testRoom,
  },
];
