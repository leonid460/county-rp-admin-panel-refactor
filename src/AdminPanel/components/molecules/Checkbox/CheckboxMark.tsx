import styled from 'styled-components';
import {colors} from 'AdminPanel/variables';

export type CheckBoxMarkProps = {
  checked: boolean;
};

const CheckBoxMark = styled.div<CheckBoxMarkProps>`
  width: 100%;
  height: 100%;
  border-radius: 2px;

  background: ${(props) => (props.checked ? colors.blue : 'white')};

  transition: all 0.3s ease;
`;

export default CheckBoxMark;
