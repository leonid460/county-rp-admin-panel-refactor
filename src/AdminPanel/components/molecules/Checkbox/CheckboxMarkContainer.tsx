import styled from 'styled-components';
import {colors} from 'AdminPanel/variables';

const CheckboxMarkContainer = styled.label`
  display: inline-block;

  width: 20px;
  height: 20px;

  padding: 2px;
  margin-right: 10px;
  border: 1px solid ${colors.gray};
  border-radius: 5px;
  background-color: white;
  box-sizing: border-box;

  outline: none;

  cursor: pointer;
`;

export default CheckboxMarkContainer;
