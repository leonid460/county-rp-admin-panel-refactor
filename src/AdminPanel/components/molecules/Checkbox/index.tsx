import React from 'react';
import CheckboxContainer from './CheckboxContainer';
import CheckboxMarkContainer from './CheckboxMarkContainer';
import CheckboxMark, {CheckBoxMarkProps} from './CheckboxMark';


type CheckboxProps = {
  className?: string
  children?: React.ReactNode
  handleCheck: (event: React.ChangeEvent<HTMLInputElement>) => void
  id: string
}

const Checkbox = (props: CheckboxProps & CheckBoxMarkProps) => {
  return <CheckboxContainer className={props.className}>
    {props.children}

    <input
      type='checkbox'
      id={props.id}
      checked={props.checked}
      onChange={props.handleCheck}
    />

    <CheckboxMarkContainer htmlFor={props.id}>
      <CheckboxMark checked={props.checked} />
    </CheckboxMarkContainer>

  </CheckboxContainer>;
};


export default Checkbox;
