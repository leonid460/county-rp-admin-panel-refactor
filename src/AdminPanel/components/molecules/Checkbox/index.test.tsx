import React from 'react';
import Checkbox from './index';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import 'jest-styled-components';
import {colors} from 'AdminPanel/variables';

describe('Testing <Checkbox />', () => {
  const checkedProps = {
    handleCheck: () => {},
    id: 'test',
    checked: true,
  };

  const uncheckedProps = {
    handleCheck: () => {},
    id: 'test',
    checked: false,
  };

  it('matches snapshot', () => {
    const component = <Checkbox {...checkedProps} />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has a blue mark when checked', () => {
    const checkbox = mount(<Checkbox {...checkedProps} />);
    const mark = checkbox.children().find('label').children();
    expect(mark).toHaveStyleRule('background', colors.blue);
  });

  it('has a white mark when not checked', () => {
    const checkbox = mount(<Checkbox {...uncheckedProps} />);
    const mark = checkbox.children().find('label').children();
    expect(mark).toHaveStyleRule('background', 'white');
  });
});
