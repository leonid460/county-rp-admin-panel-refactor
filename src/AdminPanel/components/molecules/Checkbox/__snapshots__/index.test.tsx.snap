// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Checkbox /> matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 input[type='checkbox'] {
  display: none;
  outline: none;
}

.c1 {
  display: inline-block;
  width: 20px;
  height: 20px;
  padding: 2px;
  margin-right: 10px;
  border: 1px solid #8a8a8a;
  border-radius: 5px;
  background-color: white;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
}

.c2 {
  width: 100%;
  height: 100%;
  border-radius: 2px;
  background: #0653c0;
  -webkit-transition: all .3s ease;
  transition: all .3s ease;
}

<div
  className="c0"
>
  <input
    checked={true}
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="c1"
    htmlFor="test"
  >
    <div
      checked={true}
      className="c2"
    />
  </label>
</div>
`;
